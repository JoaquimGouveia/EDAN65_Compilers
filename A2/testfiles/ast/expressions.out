Program
  List
    FuncDecl
      IdDecl ID="b"
      List
      Block
        List
    FuncDecl
      IdDecl ID="a"
      List
      Block
        List
          VarDeclStmt
            IdDecl ID="x"
            Opt
          VarDeclStmt
            IdDecl ID="y"
            Opt
              Numeral NUMERAL="2"
          VarDeclStmt
            IdDecl ID="s"
            Opt
              Mul
                Numeral NUMERAL="2"
                Numeral NUMERAL="4"
          VarAssignStmt
            IdUse ID="a"
            Sub
              Gt
                IdUse ID="b"
                IdUse ID="c"
              Numeral NUMERAL="10"
          VarAssignStmt
            IdUse ID="a"
            Mul
              Gt
                IdUse ID="b"
                IdUse ID="c"
              IdUse ID="x"
          FunctionCallStmt
            FunctionCall
              IdUse ID="b"
              List
