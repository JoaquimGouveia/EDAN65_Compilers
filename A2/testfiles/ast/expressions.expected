Program
  List
    FuncDecl
      IdDecl ID="a"
      Block
        List
          VarDeclStmt
            IdDecl ID="x"
            Opt
          VarDeclStmt
            IdDecl ID="y"
            Opt
              Numeral NUMERAL="2"
          VarDeclStmt
            IdDecl ID="s"
            Opt
              Mul
                Numeral NUMERAL="2"
                Numeral NUMERAL="4"
          VarAssignStmt
            IdUse ID="a"
            Sub
              Gt
                IdUse ID="b"
                IdUse ID="c"
              Numeral NUMERAL="10"
          VarAssignStmt
            IdUse ID="a"
            Mul
              Gt
                IdUse ID="b"
                IdUse ID="c"
              IdUse ID="x"
